doc/users/next_whats_new/boxplot_legend_support.rst

```diff
@@ -0,0 +1,36 @@
+Legend support for Boxplot
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+Boxplots now generate legend entries and can be labelled with the
+new *label* parameter. If a patch is passed to the box with ``patch_artist=True``,
+the legend gets its handle from the patch instead of the `.Line2D` object from
+the medians.
```

Comment:
```suggestion
Boxplots now support a *label* parameter to create legend entries.
```
The second sentence on the artists is a technical detail that is not relevant in this high-level overview. It's sufficient to have this in the docstring. But you could add a sentence on str vs list of str, because these enable to fundamentally different use cases of labelling.